virtualmachine/vsphere/util.go

```diff
@@ -2107,3 +2107,57 @@ func getVmsInFolder(vm *VM, folder *object.Folder, path string) (
 	}
 	return allVms, nil
 }
+
+// getDatastoreInHost: lists datastores in a host in a cluster
+func getDatastoreInHost(vm *VM, crMo *mo.ClusterComputeResource) ([]types.ManagedObjectReference, error) {
+	var (
+		err  error
+		hsMo mo.HostSystem
+	)
+	for _, host := range crMo.Host {
+		err = vm.collector.RetrieveOne(vm.ctx, host, []string{"name", "datastore"}, &hsMo)
+		if err != nil {
+			return nil, err
+		}
+		if hsMo.Name == vm.Destination.HostSystem {
+			return hsMo.Datastore, nil
+		}
+	}
+	return nil, fmt.Errorf("Host: %s not found in cluster: %s", vm.Destination.HostSystem, crMo.Name)
+}
+
+// getSharedDatastoreInCluster: get the datastores shared accross all the
+// datastores in cluster
```

Comment:
update the comment: 
"...shared accross all the
// ~~datastores~~ hosts in cluster"